src/Field/Definition/RichTextField.php

```diff
@@ -176,4 +182,169 @@ private function contentIsEmpty (array $data) : bool
 		return 0 === \count($firstItemContent) &&
 			\in_array($firstItemType, ["paragraph", "heading"], true);
 	}
+
+	private function transformRichTextContent (
+		ComponentContext $context,
+		array $richText,
+	) : array
+	{
+		$transformed = [];
+
+		foreach ($richText as $section)
+		{
+			$transformed[] = $this->transformSection($context, $section);
+		}
+
+		return $transformed;
+	}
+
+	private function transformSection (
```

Comment:
I initially had that in mind, but currently we can't do that as we're limited to one visitor per component and we're already using some visitors in most scenarios. This would require a larger refactoring. If you're willing to add this BC break (now would be the best time to do that), then I'm open about it.

Reply:
If we're going with this approach (which I'd approve), then I'd also extract the `fetchBlockDataMap()` manipulation into the visitor, as it's technically the same thing (manipulating/transforming the output) but currently done inside the `RichTextField`.